xonsh/built_ins.py

```diff
@@ -472,6 +469,9 @@ def _redirect_io(streams, r, loc=None):
     else:
         raise XonshError('Unrecognized redirection command: {}'.format(r))
 
+def known_commands():
+    return _compute_known_commands(ENV, builtins.aliases)
```

Comment:
Probably better to use `builtns.__xonsh_env__` than `ENV`.  `ENV` should really go away...